test/test_fft_unthreaded.py

```diff
@@ -50,7 +50,10 @@
     kdict = {'backends' : [backend], 'scheme' : _scheme, 'context' : _context}
     klass = type('{0}_{1}_test'.format(_scheme,backend),
                  (_BaseTestFFTClass,),kdict)
+    klass.__test__ = True
+    vars()[klass.__name__] = klass
     FFTTestClasses.append(klass)
+    del klass
```

Comment:
Surely this `del` does nothing???

Reply:
pytest works by reading what classes are defined, so in this case it actually does. Otherwise, there would be some duplicated tests run from this module.

Reply:
Hmm, I still don't see it. If I look down to next few files, they look the same and don't include a `del` here.

Reply:
@spxiwh ... that would be because I forgot to do this elsewhere, so we are doing some duplicate testing. What pytest will do is essentially treat the two named instances (the remnant 'klass' after the loop and the explicitly named one put in the local scope) as two different test sets to do and will run both.

Reply:
@spxiwh Actually, I went through and could find only one case where this might also occur in test_frame.py. Maybe you are looking at the things in __main__? That I don't need to deal with as it isn't called as pytest just does an import of the module. Everything else I see that uses the name 'klass' actually needs it so that pytest can find that one instance. 

Most of the changes I needed in fact were to enable pytest to (1) not try to run tests from a class which is a *base* class and (2) enable pytest to find our auto-generated classes that handle testing various dtype combinations for example. To do so, they all need to appear in the local scope of the module with different names (and only once to avoid being run twice)

Reply:
Okay, I think I've gotten it: is the point that because we did the weird `vars()[klass.__name__] = klass` this appears in the namespace twice unless deleted. Other cases without a loop do not need this new line and are then not a problem?? ...The vars() thing does look nasty, but the only alternative I can think of involves `exec/eval` which is probably worse.

Reply:
@spxiwh Yep, that's right. As far as I can tell var() is the only way to edit the current namespace programmatically.